api/envoy/extensions/filters/network/local_ratelimit/v3/local_rate_limit.proto

```diff
@@ -44,4 +45,9 @@ message LocalRateLimit {
   // Runtime flag that controls whether the filter is enabled or not. If not specified, defaults
   // to enabled.
   config.core.v3.RuntimeFeatureFlag runtime_enabled = 3;
+
+  // Specifies that the token bucket used for rate limiting should be shared with other local_rate_limit filters
+  // with a matching :ref:`token_bucket <envoy_v3_api_field_extensions.filters.network.local_ratelimit.v3.LocalRateLimit.token_bucket>`
+  // and `share_key` configuration. All fields of `token_bucket` must match exactly for the token bucket to be shared.
+  string share_key = 4;
```

Comment:
Will this be used also for http filters?
I'm trying to understand if this should be part of the `TokenBucket` type, as it is highly coupled with it (i.e., for filters to share this there needs to be equality of all fields in the bucket and this field). In other words, is this a property of the filter or the bucket?

Reply:
Yeah I would be concerned about use cases for TokenBucket that don't share so it seems OK to put it here, though I don't feel strongly about it. I could go either way.

Reply:
Ok, I think it makes sense.
I suggest adding to the comment what happens if share_key isn't set (e.g., If no share_key is provided each filter will have its own rate limit).

Reply:
I documented that empty means no sharing.